include/service/entities/plugin_service.js

```diff
@@ -1277,28 +1316,53 @@ module.exports = function PluginServiceModule(pb) {
-                        
-                        hasDependencies = false;
-                        return callback(null, false);
+                var rootPackagePath = path.join(process.cwd(), 'node_modules', keys[i], 'package.json');
+                var checkModulePath = function(myPath, cb) {
+                    fs.exists(myPath, function(exists) {
+                        if (!exists) {
+                            hasDependencies = false;
+                            return cb(null, {satisfied: false, exists: exists});
+                        }
+
+                        //ensure that the version expression specified by the
+                        //dependency is satisfied by that provided by the package.json
+                        var package = require(myPath);
+                        var isSatisfied = semver.satisfies(package.version, plugin.dependencies[keys[i]]);
+                        if (!isSatisfied) {
+
+                            hasDependencies = false;
+
+                        }
+                        cb(null, {satisfied: isSatisfied, exists: exists});
+                    });
+                };
+
+                checkModulePath(packagePath, function(err, result) {
+                    if(err) {
+                        return callback(err, false);
                     }
-                    
-                    //ensure that the version expression specified by the 
-                    //dependency is satisfied by that provided by the package.json
-                    var package = require(packagePath);
-                    var isSatisfied = semver.satisfies(package.version, plugin.dependencies[keys[i]]);
-                    if (!isSatisfied) {
-                        
-                        hasDependencies = false;
-                        pb.log.warn('PluginService: Plugin %s has incorrect dependency version %s for %s', plugin.name, package.version, keys[i]);
+                    if(result.satisfied) {
+                        hasDependencies = true;
+                        return callback(err, result.satisfied);
                     }
-                    callback();
+                    checkModulePath(rootPackagePath, function(err,  result) {
+                        if(err) {
+                            return callback(err, false);
+                        }
+                        if (!result.exists) {
+                            pb.log.warn('PluginService: Plugin %s is missing dependency %s', plugin.name, keys[i]);
+                        }
+                        if(!result.satisfied) {
+                            pb.log.warn('PluginService: Plugin %s has incorrect dependency version %s for %s', plugin.name, package.version, keys[i]);
+                        }
+                        if(result.satisfied) {
+                            hasDependencies = true;
+                        }
+                        callback(err, result.satisfied);
+                    });
```

Comment:
Take a look at ```async.some()```, docs are [here](https://github.com/caolan/async#some).  Should allow you to prevent the nesting.  You can just move the logic that outputs the warning logic to the ```checkModulePath``` function.